shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -0,0 +1,96 @@
+// Copyright 2018 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <map>
+#include <memory>
+#include <string>
+
+#include "FlutterPlatformViews_Internal.h"
+#include "flutter/fml/platform/darwin/scoped_nsobject.h"
+#include "flutter/shell/platform/darwin/ios/framework/Headers/FlutterChannels.h"
+
+@implementation FlutterPlatformViewsController {
```

Comment:
How is this meant to be different from a `FlutterViewController`?

I also wonder if we're further confusing some already confusing naming conventions.  We have

- `FlutterViewController`, which is a `UIViewController`
- `FlutterView` which is a `UIView`
- `PlatformViewIOS`, which is a `shell::PlatformView` (but not a view in the sense of a `UIView` or `UIViewController`

It might be better if we could extend `FlutterViewController` to do the job that this class is doing now, especially with some of the refactoring I've been working on to disentangle the engine from the view controller.

Reply:
Good point, I agree the name is confusing as this is not a ViewController but the controller for the platform views... lets see if we can come up with a better name (let me know if you have a good suggestion).

As for keeping this logic inside `FlutterViewController` vs keeping it in a separate class, we can definitely do both, I actually discussed this with Chinmay and we didn't have a really strong preference,  we leaned to encapsulating it in it's own file as FlutterViewController is starting to get a little big (at least before the coming refactor).
Keeping it out of `FlutterViewController` might also make this specific part of the logic a little easier to unit test (the only dependency a test needs to provide is a messenger in order to exercise the platform views code).